tests/integration/high_availability/test_self_healing.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+
+import pytest
+from helpers import get_primary_unit, get_process_pid
+from pytest_operator.plugin import OpsTest
+from tenacity import Retrying, stop_after_delay, wait_fixed
+
+from tests.integration.high_availability.high_availability_helpers import (
+    clean_up_database_and_table,
+    ensure_all_units_continuous_writes_incrementing,
+    ensure_n_online_mysql_members,
+    get_process_stat,
+    high_availability_test_setup,
+    insert_data_into_mysql_and_validate_replication,
+    send_signal_to_pod_container_process,
```

Comment:
As you keep writing these tests you will start creating more functions and importing more. it might be nice to change to 
```
import  high_availability_helpers

high_availability_helpers.get_process_stat
```
but fully up to you

Reply:
since we're importing `tests.integration.high_availability.high_availability_helpers`, i'll keep as is for now (also for consistency in other imports in test files). will keep this in mind for the future however!